### ISSUE TITLE
Existing orders break if the plugin of the payment provider gets disabled

### ISSUE THREAD

**Issue by raphaelm** [pretix/pretix#66]

**Comment by lhhong** [comment]
It seemed like there are two ways to go about this problem:

1. Fix the bug as stated to make the orders still work when the plugin gets disabled
2. Do not allow users to disable the plugin if there are already payments made

I feel that option 2 seemed to be a better option as it is much easier to fix (correct me if I am wrong as I have not looked at the implementation of plugins), without violating intuitiveness of user experience. With sufficient feedback message, it is natural to a user to not be able to disable the plugin. Furthermore, the user can still un-check the "enable" box under payment settings if they do not wish to receive further payment based on it.

**Comment by raphaelm** [comment]
> I feel that option 2 seemed to be a better option as it is much easier to fix (correct me if I am wrong as I have not looked at the implementation of plugins), without violating intuitiveness of user experience.

Yes and no. We can do this, sure, but we can't keep a system administrator from *uninstalling* the plugin globally (i.e. deleting it), at least in case of plugins that ship individually from pretix. So in the end, we probably need *both* option 1 and 2.
